Firestore/core/src/firebase/firestore/remote/grpc_stream_operation.h

```diff
@@ -48,16 +48,12 @@ class GrpcStream;
  * Operation expects all GRPC objects pertaining to the current stream to remain
  * valid until the operation comes back from the GRPC completion queue.
  */
-class StreamOperation : public GrpcOperation {
+class GrpcStreamCompletion : public GrpcOperation {
  public:
-  explicit StreamOperation(GrpcStream* stream);
+  using Completion = std::function<void(bool, const GrpcStreamCompletion&)>;
```

Comment:
Maybe call this `Action` or `CompletionAction` to disambiguate the function we'll call from this type that's a holder for the things related to that function?